src/common/src/array/data_chunk_iter.rs

```diff
@@ -315,6 +315,13 @@ impl Row {
         }
         Ok(HashCode(hasher.finish()))
     }
+
+    /// Get an owned `Row` by the given `indices` from current row.
+    ///
+    /// Use `datum_refs_by_indices` if possible instead to avoid allocating owned datums.
+    pub fn by_indices(&self, indices: &[usize]) -> Row {
```

Comment:
Off the topic: We should really review the current implementation of `serialize` functions. In most cases there's no need to allocate a `Row`.